activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -152,11 +152,10 @@ def preventing_writes?
       # even if you are on a database that can write. `while_preventing_writes`
       # will prevent writes to the database for the duration of the block.
       def while_preventing_writes
-        original = self.prevent_writes
         self.prevent_writes = true
         yield
       ensure
-        self.prevent_writes = original
+        self.prevent_writes = false
```

Comment:
This would make nested calls to `while_preventing_writes` surprising...

(OTOH, if the existing behaviour is worth preserving, I guess it'd also be worth a test.)

Reply:
In GH's version we actually pass in true/false as an argument to the method which I thought was weird. Now I'm thinking that maybe it's not weird. Will revert this change and work on some tests to figure out what behavior I actually want.